File: samples/vmware/pyvmomi/vsphere.py
#!/usr/bin/env python

"""
Connect to an ESX / vCenter host
"""

from lib.args import get_args
from lib.connect import connect
from lib.inventory import get_datacenter

def main():
    global datacenter

    args = get_args()

    si = connect(args.host, args.user, args.password, int(args.port))
    if not si:
        print("Error: Could not connect to the specified host using specified "
              "username and password")
        return -1

    datacenter = get_datacenter(si)
    print('datacenter: %s' % datacenter.name)
    return 0

# Start program
if __name__ == "__main__":
    main()

File: samples/vmware/rbvmomi/vsphere.rb
# Connect to VMware vSphere (vCenter Server or standalone ESX host).
# Get an interactive session with a datacenter object called dc:
#
#   pry -Ilib vsphere.rb
#   [1] pry(main)> dc.name
#   => "ha-datacenter"

require 'connect'
require 'pry'

host     = ENV['vsphere_host'] || '192.168.99.100'
user     = ENV['vsphere_user'] || 'root'
password = ENV['vsphere_password'] || raise('vsphere_password not set')

service_instance = Connect.login(host, user, password)
dc = service_instance.find_datacenter('ha-datacenter') || raise('datacenter not found')

puts "Datacenter [dc]: #{dc.name}"

Pry.start
